src/ConnectedMode.UnitTests/Persistence/LegacySolutionBindingPathProviderTests.cs

```diff
@@ -19,67 +19,58 @@
  */
 
 using System;
-using System.IO;
-using Microsoft.VisualStudio.Shell.Interop;
 using SonarLint.VisualStudio.ConnectedMode.Persistence;
+using SonarLint.VisualStudio.Core;
 
 namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Persistence
 {
     [TestClass]
     public class LegacySolutionBindingPathProviderTests
     {
-        private Mock<IVsSolution> solution;
-        private LegacySolutionBindingPathProvider testSubject;
-
-        [TestInitialize]
-        public void TestInitialize()
-        {
-            solution = new Mock<IVsSolution>();
-
-            var serviceProvider = new Mock<IServiceProvider>();
-            serviceProvider.Setup(x => x.GetService(typeof(SVsSolution))).Returns(solution.Object);
-
-            testSubject = new LegacySolutionBindingPathProvider(serviceProvider.Object);
-        }
-
         [TestMethod]
         public void Ctor_NullArgument_Exception()
         {
             Action act = () => new LegacySolutionBindingPathProvider(null);
 
-            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("serviceProvider");
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("solutionInfoProvider");
         }
 
         [TestMethod]
-        public void Get_SolutionFolderIsNull_Null()
+        public void Ctor_DoesNotCallServices()
```

Comment:
New test that the constructor doesn't switch threads/call anything that might switch threads.